### DISCUSSION TITLE
Ondemand ISR - clears cache of all pages

### DISCUSSION THREAD

**Discussion by miguelvictor** [vercel/next.js#41604] [Ideas]
### Describe the feature you'd like to request

An API similar to `revalidate` but clears the cache for all pages instead of just a single page without needing to rebuild the entire web app.

### Describe the solution you'd like

Currently, we have "global fields" in our CMS backend. After we update some of the fields there, our users are expecting to see the updated values in all of the pages within our site. So, an API like `revalidate` is desirable here.

### Describe alternatives you've considered

We try to get all of the static paths from all of the pages and combine them together in a single array. We then call the `revalidate` function wrapped in `Promise.all` for concurrency.

**Comment by JB-Scott** [comment]
We also support the addition of this feature.

**Our Issue**
We need to update all pages once a day at a particular time. Setting a revalidate time frame in getStaticProps() is not going to guarantee the revalidate happens at the correct time. The alternative of calling every page with the revalidate() function is too error prone when pages are being added or removed over time. 

**Current Solution**
The only viable solution for us right now is to completely rebuild the site at that specific time each day.

A single on-demand revalidate() function that updates all pages is greatly preferred and would be faster than a full site rebuild.

**Alternate Solution**
While, the above suggestion would work for us, an alternate solution could involve another setting in the getStaticProps() function. If you could specify a time each day that signals revalidation should occur, then the first call that comes in after that time on that day would invalidate the page and that would give us the desired result.
